code/game/atoms.dm

```diff
@@ -353,6 +353,8 @@ var/list/blood_splatter_icons = list()
 /atom/proc/add_vomit_floor(mob/living/carbon/M as mob, var/toxvomit = 0)
 	if( istype(src, /turf/simulated) )
 		var/obj/effect/decal/cleanable/vomit/this = new /obj/effect/decal/cleanable/vomit(src)
+		if(M.reagents)
+			M.reagents.trans_to(this, M.reagents.total_volume / 10)

```

Comment:
I don't think "this" is a BYOND keyword. Maybe you meant "src", which is the equivalent.